api/src/main/java/org/openmrs/api/db/hibernate/PatientSearchCriteria.java

```diff
@@ -467,116 +498,146 @@ String[] getQueryParts(String query) {
+		String pattern = getMatchMode().toCaseSensitivePattern(name);
+
+		Predicate givenNamePredicate = cb.like(nameJoin.get("givenName"), pattern);
+		Predicate middleNamePredicate = cb.like(nameJoin.get("middleName"), pattern);
+		Predicate familyNamePredicate = cb.like(nameJoin.get("familyName"), pattern);
+		Predicate familyName2Predicate = cb.like(nameJoin.get("familyName2"), pattern);
+
+		Predicate namePredicate = cb.or(givenNamePredicate, middleNamePredicate, familyNamePredicate, familyName2Predicate);
+
 		if (!includeVoided) {
-			return Restrictions.conjunction().add(Restrictions.eq("name.voided", false)).add(criterion);
+			Predicate nonVoidedPredicate = cb.isFalse(nameJoin.get("voided"));
+			namePredicate = cb.and(namePredicate, nonVoidedPredicate);
 		}
-		return criterion;
+
+		return namePredicate;
 	}
-	
-	private Criterion getCriterionForNoExactName(String name, boolean includeVoided) {
-		MatchMode matchMode = getMatchMode();
-		
-		Criterion criterion = Restrictions.conjunction().add(
-		    Restrictions.disjunction().add(
-		        Restrictions.conjunction().add(Restrictions.isNotNull("name.givenName")).add(
-		            Restrictions.like("name.givenName", name, matchMode))).add(
-		        Restrictions.conjunction().add(Restrictions.isNotNull("name.middleName")).add(
-		            Restrictions.like("name.middleName", name, matchMode))).add(
-		        Restrictions.conjunction().add(Restrictions.isNotNull("name.familyName")).add(
-		            Restrictions.like("name.familyName", name, matchMode))).add(
-		        Restrictions.conjunction().add(Restrictions.isNotNull("name.familyName2")).add(
-		            Restrictions.like("name.familyName2", name, matchMode)))).add(
-		    Restrictions.disjunction().add(Restrictions.isNull("name.givenName")).add(
-		        Restrictions.ne("name.givenName", name))).add(
-		    Restrictions.disjunction().add(Restrictions.isNull("name.middleName")).add(
-		        Restrictions.ne("name.middleName", name))).add(
-		    Restrictions.disjunction().add(Restrictions.isNull("name.familyName")).add(
-		        Restrictions.ne("name.familyName", name))).add(
-		    Restrictions.disjunction().add(Restrictions.isNull("name.familyName2")).add(
-		        Restrictions.ne("name.familyName2", name)));
-		
+
+	private Predicate getPredicateForNoExactName(CriteriaBuilder cb, Join<Patient, PersonName> nameJoin, String name, boolean includeVoided) {
+		String pattern = getMatchMode().toCaseSensitivePattern(name);
+		Predicate givenNamePredicate = cb.and(
+			cb.isNotNull(nameJoin.get("givenName")),
+			cb.like(cb.lower(nameJoin.get("givenName")), pattern)
+		);
+		Predicate middleNamePredicate = cb.and(
+			cb.isNotNull(nameJoin.get("middleName")),
+			cb.like(cb.lower(nameJoin.get("middleName")), pattern)
+		);
+		Predicate familyNamePredicate = cb.and(
+			cb.isNotNull(nameJoin.get("familyName")),
+			cb.like(cb.lower(nameJoin.get("familyName")), pattern)
```

Comment:
Should we really use `lower()` for this one?

Reply:
Fixed